### ISSUE TITLE
fileReplacements no longer work with react and webpack

### ISSUE THREAD

**Issue by a88zach** [nrwl/nx#20192]
### Current Behavior

When building a react app using @nx/webpack:webpack, file replacements are no longer working.  Looks like this commit https://github.com/nrwl/nx/commit/395eb70336cb1346548705d7c41d896b4a6c6cbb is the culprit 

### Expected Behavior

file replacements should work when building or serving a react application

### Steps to Reproduce

1.  Set up a nx workspace with a react app. 
2. Create a production config using a file replacement
3. Build the production config
4. Notice that the file replacements do not work

### Nx Report

```shell
Node   : 20.9.0
   OS     : darwin-arm64
   yarn   : 3.6.4
   
   nx                 : 17.1.1
   @nx/js             : 17.1.1
   @nx/jest           : 17.1.1
   @nx/linter         : 17.1.1
   @nx/eslint         : 17.1.1
   @nrwl/linter       : 15.9.2
   @nx/workspace      : 17.1.1
   @nx/cypress        : 17.1.1
   @nx/devkit         : 17.1.1
   @nx/esbuild        : 17.1.1
   @nx/eslint-plugin  : 17.1.1
   @nx/node           : 17.1.1
   @nx/react          : 17.1.1
   @nx/storybook      : 17.1.1
   @nrwl/tao          : 17.1.1
   @nx/vite           : 17.1.1
   @nx/web            : 17.1.1
   @nx/webpack        : 17.1.1
   typescript         : 5.2.2
```

### Operating System

- [X] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by a88zach** [comment]
cc @jaysoo

**Comment by Burgov** [comment]
I can confirm this is an issue. Downgrading to 17.0.* fixed it for me.

**Comment by ericmartinezr** [comment]
Not entirely sure if related to this same issue reported but outputPath stopped working as well

**Comment by yharaskrik** [comment]
I am experiencing the same thing, just with a Nest app.

**Comment by kilotone** [comment]
Only experience it when building on non windows,  I also didn't use a webpack config for this, just  options and config from project.json

**Comment by yharaskrik** [comment]
Seems like it works if using relative paths like this

```json
                    "fileReplacements": [
                        {
                            "replace": "./environments/environment",
                            "with": "./environments/environment.master.ts"
                        }
                    ]
```

**Comment by a88zach** [comment]
Unfortunately, using relative paths is still not working.

**Comment by KeithGillette** [comment]
I believe we have encountered this issue upon migrating from `@nx@17.02` to `nx@17.1.2` with a NodeJS application built with `@nrwl/webpack:webpack` and served with `@nx/js:node`, as environment variable values do not appear to be available at run unless the values are manually entered into `environment.ts`.

```
 >  NX   Report complete - copy this into the issue template

   Node   : 18.16.1
   OS     : darwin-arm64
   npm    : 9.5.1
   
   nx             : 17.1.2
   @nx/js         : 17.1.2
   @nx/jest       : 17.1.2
   @nx/linter     : 17.1.2
   @nx/eslint     : 17.1.2
   @nx/workspace  : 17.1.2
   @nx/angular    : 17.1.2
   @nx/cypress    : 17.1.2
   @nx/devkit     : 16.5.1
   @nx/node       : 17.1.2
   @nrwl/tao      : 17.1.2
   @nx/web        : 17.1.2
   @nx/webpack    : 17.1.2
   typescript     : 5.2.2
   ---------------------------------------
   Community plugins:
   apollo-angular : 5.0.2
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nx/devkit@16.5.1
     - @nrwl/devkit@16.5.1
   ```

(`@nx/devkit` & `@nrwl/devkit` are not in our `package.json`, so I'm not sure why they keep showing up even after deleting `node_modules` and running `npm install`

**Comment by kilotone** [comment]
I actually might just be experiencing a side effect, I recently upgraded nx
to 17.0.2 and had to switch both my electron and express app to use the
webpack:webpack executor.   The file replacements work fine when I dont
have the target as node, but, if I do, then it silently proceeds on build,
and does not replace file.
This also seems it could be related to thr use of alpine linuxz as the
replacements work fine on windows.

**Comment by vadimka123** [comment]
@jaysoo , any update? 
It’s critical blocker for use last NX versions on live environments

**Comment by guitarooman14** [comment]
Like mention in the issue [20253](https://github.com/nrwl/nx/issues/20253), relative paths are no more working. Absolute paths does, but it is not something that I can commit. Thanksfully I'm not yet in production ...

**Comment by tsmithhisler** [comment]
Experiencing this after upgrading to v17.1.x. I suspect this has something to do with the following line from a commit a few weeks ago: https://github.com/nrwl/nx/commit/395eb70336cb1346548705d7c41d896b4a6c6cbb#diff-891ed1c6cd733eaa28d9f44efe10195ee093b21f3fec16330217ef78d21799cdL28

**Comment by julianpoemp** [comment]
same issue. Downgrading to 17.0.3 fixed it for me

**Comment by deser** [comment]
@AgentEnder  Any ETA on the fix?

**Comment by Bielik20** [comment]
May be related. We are not using fileReplacements but our builds produce _corrupted_ result. We are using webpack executor to bundle lambdas for AWS. Since 17.1.x deployed lambdas do not work, so we know that output is somehow corrupted.

**Comment by AdaskoTheBeAsT** [comment]
@jaysoo could you please help with that issue?

**Comment by kfrederix** [comment]
Observed the same here: running build with `@nx/webpack` executor results in artifacts getting created under duplicate "dist" folder. ( e.g. C:\my_repo\dist\dist\libs\... instead of C:\my_repo\dist\libs\...)

**Comment by wSedlacek** [comment]
> Seems like it works if using relative paths like this
> 
> ```json
>                     "fileReplacements": [
>                         {
>                             "replace": "./environments/environment",
>                             "with": "./environments/environment.master.ts"
>                         }
>                     ]
> ```

So it appears this format works IF your import looks like `import {} from './environments/environment';` however if you use a barrel in `environments/index.ts` this format doesn't work and you need to use something like this.
```json
            {
              "replace": "./environment",
              "with": "./environment.prod.ts"
            },
```

Also note: The `replace` does NOT have `.ts`

**Comment by gorkemyontem** [comment]
👋 Hey Nx team,

I want to express the urgency and critical nature of this issue. The problem with file replacements not working in React or Nestjs applications using @nx/webpack:webpack, as highlighted in this thread, is having a severe impact on multiple users, hindering their build and serve processes.

🔍 Observations:
- File replacements are failing after commit 395eb70.
- Downgrading to 17.0.* resolves the issue for some users.
- Impact extends to outputPath, webpack file paths, and other related concerns.

🌐 Community Impact:
- Blocking deployments for users in live environments.
- Reverting to older versions as a temporary solution.
- High risk for projects dependent on correct file replacements.

👥 Community Confirmation:
- Multiple users have confirmed the issue.
- Impact extends to various projects, including React and Nest applications.

📅 Timeline:
- The issue has been open for two weeks without a resolution.
- Users are seeking updates and an ETA on the fix.

🙏 Request:
- Prioritize the resolution of this critical bug.
- Provide guidance on potential workarounds.
- Keep the community updated on the progress.

This bug is affecting a substantial number of users and has become a blocker for those relying on the latest Nx versions in production. Addressing this promptly will have a positive impact on the community and their projects.

Thank you for your attention to this matter.

**Comment by jacqueslareau** [comment]
still not working? what is the workaround??

**Comment by KeithGillette** [comment]
I think we're waiting on the release of `nx@17.2.0`, @jacqueslareau, as a fix closing the issue was merged into [`nx@17.20-beta.8`](https://github.com/nrwl/nx/releases/tag/17.2.0-beta.8), though I have not personally tested.

**Comment by jacqueslareau** [comment]
Issue still present in 17.2.1

**Comment by jacqueslareau** [comment]
Nevermind. I had to revert changes I made in the file to try to find a workaround. It's working now. Thank you!

**Comment by tylerwalts** [comment]
I was experiencing this issue where a React package's environment.ts production build `fileReplacements` was not working while using nx `17.1.3`, but can confirm that it has been resolved after upgrading all `@nx*` references in root package.json to `17.2.8` (latest as of today).
